packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-checkbox-group/gux-form-field-checkbox-group.tsx

```diff
@@ -0,0 +1,238 @@
+import {
+  Component,
+  Element,
+  forceUpdate,
+  h,
+  JSX,
+  State,
+  Prop,
+  Watch
+} from '@stencil/core';
+
+import { OnMutation } from '@utils/decorator/on-mutation';
+import { hasSlot } from '@utils/dom/has-slot';
+import { logWarn } from '@utils/error/log-error';
+
+import {
+  setAllCheckboxInputs,
+  setMainCheckboxElementCheckedState
+} from './gux-form-field-checkbox-group.service';
+
+import {
+  GuxFormFieldError,
+  GuxFormFieldHelp,
+  GuxFormFieldLegendLabel,
+  GuxFormFieldFieldsetContainer
+} from '../../functional-components/functional-components';
+import { getSlotTextContent } from '@utils/dom/get-slot-text-content';
+
+import { trackComponent } from '@utils/tracking/usage';
+
+/**
+ * @slot group-label - Required slot for label tag
+ * @slot group-checkbox - Optional slot
+ * @slot group-error - Optional slot for error message
+ * @slot group-help - Optional slot for help message
+ */
+@Component({
+  styleUrl: 'gux-form-field-checkbox-group.scss',
+  tag: 'gux-form-field-checkbox-group-beta',
+  shadow: true
+})
+export class GuxFormFieldCheckboxGroupBeta {
+  private disabledObserver: MutationObserver;
+  private label: HTMLLabelElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  /**
+   *  Checkbox group has error text.
+   */
+  @State()
+  private hasGroupError: boolean = false;
+
+  /**
+   *  Checkbox group has help text.
+   */
+  @State()
+  private hasGroupHelp: boolean = false;
+
+  /**
+   * Disables the checkboxes in the group.
+   */
+  @Prop()
+  disabled: boolean = false;
+
+  @Watch('hasGroupError')
+  watchGroupError(hasGroupError: boolean): void {
+    const checkboxSlots = this.root.querySelectorAll('gux-form-field-checkbox');
+    if (checkboxSlots) {
+      checkboxSlots.forEach(item => {
+        item.hasGroupError = hasGroupError;
+      });
+    }
+  }
+
+  @Watch('disabled')
+  watchDisabled(): void {
+    this.setDisabledCheckbox();
```

Comment:
Are we setting one single checkbox or multiple checkbox disabled states? I think it's multiple but wanted to confirm. If it's multiple could we change it to plural in the method name?